packages/react-aria-components/docs/Virtualizer.mdx

```diff
@@ -0,0 +1,430 @@
+      <ListBox aria-label="Virtualized ListBox" selectionMode="multiple" items={items}>
+        {item => <ListBoxItem>{item.name}</ListBoxItem>}
+      </ListBox>
+    </Virtualizer>
+  );
+}
+```
+
+## Features
+
+Virtualized scrolling is a performance optimization for large lists. Instead of rendering all items to the DOM at once, it only renders the visible items, reusing them as the user scrolls. This results in a small number of DOM elements being rendered, reducing memory usage and improving browser layout and rendering performance.
+
+* **Integrated** – Works with React Aria [ListBox](ListBox.html), [GridList](GridList.html), and [Table](Table.html) components. Integrated with React Aria's [drag and drop](dnd.html), [selection](selection.html), and [table column resizing](Table.html#column-resizing) implementations.
+* **Custom layouts** – Support for list, grid, and table layouts out of the box, with fixed or variable size items. Create a <TypeLink links={docs.links} type={docs.exports.Layout} /> subclass to build your own custom layout.
+* **Accessible** – Persists the focused element in the DOM even when out of view, ensuring keyboard navigation always works. Adds ARIA attributes like `aria-rowindex` to give screen reader users context.
+
+## Anatomy
+
+Collection components such as [ListBox](ListBox.html), [GridList](GridList.html), and [Table](Table.html) can be virtualized by wrapping them in a &lt;<TypeLink links={docs.links} type={docs.exports.Virtualizer} />&gt;, and providing a <TypeLink links={docs.links} type={docs.exports.Layout} /> object such as <TypeLink links={docs.links} type={docs.exports.ListLayout} /> or <TypeLink links={docs.links} type={docs.exports.GridLayout} />. See below for examples of each layout.
+
+```tsx
+import {Virtualizer, ListLayout} from 'react-aria-components';
+
+let layout = useMemo(() => new ListLayout(), []);
+
+<Virtualizer layout={layout}>
+  <ListBox>
+    {/* ... */}
+  </ListBox>
+</Virtualizer>
+```
+
+<InlineAlert variant="notice" maxWidth={600}>
+  <Heading>Important Requirements</Heading>
+  <Content>
+    <ol style={{margin: 0, paddingLeft: '1em'}}>
+      <li><strong>Layout objects must be memoized.</strong> Layouts store cached data that must not be reset on every React render.</li>
+      <li><strong>Virtualized components must have a defined size.</strong> This may be an explicit CSS `width` and `height`, or an implicit size (e.g. percentage or `flex`) bounded by an ancestor element. Without a bounded size, all items will be rendered to the DOM, negating the performance benefits of virtualized scrolling.</li>
+    </ol>
+  </Content>
+</InlineAlert>
+
+## Layouts
+
+Virtualizer uses <TypeLink links={docs.links} type={docs.exports.Layout} /> objects to determine the position and size of each item, and provide the list of currently visible items. When using a Virtualizer, all items are positioned by the `Layout`, and CSS layout properties such as flexbox and grid do not apply.
```

Comment:
```suggestion
Virtualizer uses <TypeLink links={docs.links} type={docs.exports.Layout} /> objects to determine the position and size of each item, and provides the list of currently visible items. When using a Virtualizer, all items are positioned by the `Layout`, and CSS layout properties such as flexbox and grid do not apply.
```

Reply:
I think provide is correct. "Virtualizer uses Layout objects to ... provide the list of currently visible items"